draft-ietf-quic-transport.md

```diff
@@ -2320,15 +2362,28 @@ Sequence:
   the value selected during the handshake comes immediately before the first
   value that a server can send.
 
+Length:
+
+: An 8-bit unsigned integer containing the length of the connection ID.  Values
+  less than 8 and greater than 22 are invalid and MUST be treated as a
+  connection error of type PROTOCOL_VIOLATION.
```

Comment:
What is actually the use case for changing the connection ID length during a connection? Is that really needed?

Reply:
Well, if you are migrating to new load balancing infrastructure, it might help.  Note that you probably need an indicator of length in the connection ID itself to make this work, or a period where the common prefix is distinct.

Basically, I couldn't see any reason not to allow this, given that it's the endpoint that sends NEW_CONNECTION_ID that absorbs the complexity cost.